pkg/services/ngalert/remote/alertmanager_test.go

```diff
@@ -137,20 +137,24 @@ func TestApplyConfig(t *testing.T) {
 	})
 
 	var configSent client.UserGrafanaConfig
-	var lastConfigSync, lastStateSync time.Time
+	var configSyncs, stateSyncs int
 	okHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		require.Equal(t, tenantID, r.Header.Get(client.MimirTenantHeader))
 		require.Equal(t, "true", r.Header.Get(client.RemoteAlertmanagerHeader))
+		var res = map[string]any{"status": "success"}
+
 		if r.Method == http.MethodPost {
 			if strings.Contains(r.URL.Path, "/config") {
 				require.NoError(t, json.NewDecoder(r.Body).Decode(&configSent))
-				lastConfigSync = time.Now()
+				configSyncs++
 			} else {
-				lastStateSync = time.Now()
+				stateSyncs++
 			}
+		} else {
+			res["data"] = &configSent
```

Comment:
I mocked the remote AM sending back the latest config.
Previously, the handler returned nothing, so all comparisons had the same outcome.